File: other/Solution69.java
class Solution69 {
    public int mySqrt(int x) {
    	long result = 0;
    	while(x >= result * result){
    		result += 1;
    	}
    	return (int) result - 1;
    }

    public static void main(String[] args){
    	Solution69 s = new Solution69();
    	System.out.println(s.mySqrt(2147395600));
    }
}

File: C++/Solution69.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
using namespace std;

class Solution69 {
public:
    int mySqrt(int x) {
        if(x <= 1){
            return x;
        }
        int lower = 1, upper = x;
        while(upper - lower > 1){
            int mid = (upper - lower) / 2 + lower;
            long long middle = mid;
            middle = middle * middle;
            if(middle == x){
                return mid;
            }
            if(middle < x){
                lower = mid;
            }
            else{
                upper = mid;
            }
        }
        return lower;
    }
};